go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/ipfs/go-ipfs-pinner
 go 1.19
 
 require (
+	github.com/Jorropo/channel v0.0.0-20230303124104-2821e25e07ff
```

Comment:
can we not add this dependency?

Reply:
If we want to return a union of "val or err" from a chan, then it's perfectly idiomatic to return a `struct { Pin coreiface.Pin, Err error}`, I'd prefer not to add another module to our rat's nest of dependencies for something that already has an established idiom.

Reply:
> has an established idiom

I don't belive that true, most code in the std is synchronous or uses methods of interface who secretly manage synchronisation.

Reply:
We don't need a dependency for this. Please don't do it